src/main/java/com/testjava/metrics_platform/TestJavaEventSender.java

```diff
@@ -18,38 +23,40 @@
 import com.google.gson.Gson;
 
 public class TestJavaEventSender implements EventSender {
+
+    private final Gson gson = new Gson();
+
     @Override
-    public void sendEvents(String baseUri, Collection<Event> events) throws MalformedURLException {
+    public void sendEvents(String baseUri, Collection<Event> events) throws IOException {
```

Comment:
throwing IOException is a change of the EventSender interface. I think that's a required change, the caller needs to be able to know if events were sent or if there was an error.